.github/actions/bumpVersion/bumpVersion.js

```diff
@@ -1,71 +1,55 @@
 
-                // Get the highest build version git tag from the repo
-                console.log('Fetching tags from github...');
-                const octokit = github.getOctokit(core.getInput('GITHUB_TOKEN', {required: true}));
-                octokit.repos.listTags({
-                    owner: repoOwner,
-                    repo: repoName,
-                })
-                    .then((response) => {
-                        const tags = response.data.map(tag => tag.name);
-                        console.log('Tags: ', tags);
-                        const highestBuildNumber = Math.max(
-                            ...(tags
-                                .filter(tag => tag.startsWith(currentPatchVersion))
-                                .map(tag => tag.split('-')[1])
-                            ),
-                        );
-                        console.log('Highest build number from current patch version:', highestBuildNumber);
+// SEMVER_LEVEL defaults to BUILD
+// it actually would fall to build anyway, but I think it is better to make it explicitly
+if (!semanticVersionLevel || !Object.values(functions.semanticVersionLevels).find(v => v === semanticVersionLevel)) {
+    // eslint-disable-next-line max-len, semi
+    console.log(`Invalid input for 'SEMVER_LEVEL': ${semanticVersionLevel}, defaulting to: ${functions.semanticVersionLevels.build}`)
+    semanticVersionLevel = functions.semanticVersionLevels.build;
+}
 
-                        const newBuildNumber = `${currentPatchVersion}-${highestBuildNumber + 1}`;
-                        console.log(`Setting npm version for this PR to ${newBuildNumber}`);
-                        exec(`npm version ${newBuildNumber} -m "Update version to ${newBuildNumber}"`,
-                            // eslint-disable-next-line no-shadow
-                            (err, stdout, stderr) => {
-                                console.log(stdout);
-                                if (err) {
-                                    console.log(stderr);
-                                }
-                            });
-                    })
-                    .catch(exception => core.setFailed(exception));
-            } else {
-                core.setFailed(err);
-            }
-        }
+// eslint-disable-next-line no-shadow
+const bumpVersion = async (core, github, semanticVersionLevel) => {
+    console.log('Fetching tags from github...');
+    const response = await octokit.repos.listTags({
+        owner: github.context.repo.owner,
+        repo: github.context.repo.repo,
     });
-} while (shouldRetry);
+    const tags = response.data.map(tag => tag.name);
+
+    // tags come from latest to oldest
+    const highestVersion = tags[0];
+    console.log(`Highest version found: ${highestVersion}.`);
+
+    const newVersion = functions.incrementVersion(highestVersion, semanticVersionLevel);
+
+    core.setOutput('VERSION', newVersion);
```

Comment:
This output should be listed in the actions metadata file. I called it `newVersion`, but maybe it's a good idea to standardize on `UPPER_SNAKE_CASE` for GH Actions I/O

Reply:
I started working on a cloned, private repository and this output `
outputs:
    VERSION:
        description: 'New version'` got lost there even though I mention it in the version.yml
        
    I just added it to the cloned version.